Cargo.toml

```diff
@@ -66,6 +66,7 @@ itertools = "0.13"
 log = "^0.4"
 mockito = "^1"
 murmur3 = "0.5.2"
+num_cpus = "1"
```

Comment:
We can replace this by [std::thread::available_parallelism](https://doc.rust-lang.org/stable/std/thread/fn.available_parallelism.html)